### PR TITLE
Local Camera Dialog created

### PR DESCRIPTION
**Pull request by stefanodellosa-personal**
- Created new dialog to seclect cameras among a list of local cameras connected
- Added simple motion detection logic for cameras that do not provide it.

### CODE REVIEW

File: src/domain/select_local_cameras.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            col = col+1
+
+        row = 1
+        for camera_info in enumerate_cameras(cv2.CAP_MSMF): 
+            #TODO: check API preference for linux and implement it OS dependent.
+            camera_index = str(camera_info.index)
+          
+            # Camera selection check box
+            check_box = QCheckBox()
+            checkbox_obj_name = "checkBox_camera"+camera_index
```
Comment (cdimauro) [inline src/domain/select_local_cameras.py:47]: Spaces around binary operators. Also below, on all other occurrences.

File: src/domain/select_local_cameras.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            check_box.setObjectName(checkbox_obj_name)
+            check_box.checkStateChanged.connect(self.validate_cameras_selection)
+            self.ui.gridLayout_localCameras.addWidget(check_box, row, 0)
+             # Camera index
+            label = QLabel(f'{camera_info.index}')
```
Comment (cdimauro) [inline src/domain/select_local_cameras.py:52]: You already have the argument as a string: camera_index

File: src/domain/select_local_cameras.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            # Camera name
+            label = QLabel(f'{camera_info.name}')
+            label.setObjectName("label_cameraName"+camera_index)
+            self.ui.gridLayout_localCameras.addWidget(label, row, 2)
+            # Camera PID
+            label = QLabel(f'{camera_info.pid}')
+            label.setObjectName("label_cameraPid"+camera_index)
+            self.ui.gridLayout_localCameras.addWidget(label, row, 3)
+            # Camera VID
+            label = QLabel(f'{camera_info.vid}')
```
Comment (cdimauro) [inline src/domain/select_local_cameras.py:64]: Better to use str() instead of f-string on those cases.

File: src/domain/select_local_cameras.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            id_line_edit.setObjectName(lnedit_obj_name)
+            id_line_edit.textChanged.connect(self.validate_cameras_selection)
+            self.ui.gridLayout_localCameras.addWidget(id_line_edit, row, 5)
+            # Test button to preview video (and detection if enabled)
+            test_button = QPushButton('Test video')
+            test_button.setObjectName("button_camera"+camera_index)
+            self.ui.gridLayout_localCameras.addWidget(test_button, row, 6)
+            #TODO: connect slot test_button.clicked.connect()
+
+            row = row +1
```
Comment (cdimauro) [inline src/domain/select_local_cameras.py:79]: row += 1

File: src/domain/select_local_cameras.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        self.ui.gridLayout_localCameras.setAlignment(Qt.AlignmentFlag.AlignTop)
+
+    def initialize_usb_cameras(self):
+        """Method to initialize attributes from existing camera configuration (if any)."""
+        if not self.cameras_list:
+            return
+ 
+        # Initialize camera checkboxes (if any was selected before) and IDs
+        camera_number = len(enumerate_cameras(cv2.CAP_MSMF))
```
Comment (cdimauro) [inline src/domain/select_local_cameras.py:88]: Maybe num_cameras could be a better identifier

File: src/domain/select_local_cameras.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        for i in range(camera_number):
+            checkbox_obj_name = "checkBox_camera"+i
+            id_line_edit_name = "label_cameraName"+i
```
Comment (cdimauro) [inline src/domain/select_local_cameras.py:91]: You can't + strings and numbers. Better to use an f-string here.

File: src/domain/select_local_cameras.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+        valid = True
+        for camera_info in enumerate_cameras(cv2.CAP_MSMF):
+            camera_index = str(camera_info.index)
+            checkbox_obj_name = "checkBox_camera"+camera_index
+            id_line_edit_name = "label_cameraName"+camera_index
+            checkbox = self.ui.formLayout_caleraList.findChild(QCheckBox, checkbox_obj_name)
+            line_edit = self.ui.formLayout_caleraList.findChild(QLineEdit, id_line_edit_name)
+            if checkbox and checkbox.isChecked() and not line_edit.text():
+                self.ui.label_errorMessage.setText("Please add ID for Camera %s", camera_index)
```
Comment (cdimauro) [inline src/domain/select_local_cameras.py:118]: There's no string interpolation for this: "Please add ID for Camera %s", camera_index
Is it something which the method does internally, like the logging methods?

File: src/requirements.txt [hunk 8]
```diff
@@ -1,3 +1,5 @@
+opencv-python
+cv2_enumerate_cameras
 ipykernel
 timm
 keyring
```
Comment (cdimauro) [inline src/requirements.txt:5]: cv2_enumerate_cameras
ipykernel
keyring
opencv-python
timm

File: src/domain/camera.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+        """Method to run motion detection on camera video stream.
+           Only for cameras that are note povided with commercial motion detection feature."""
+
+        logger.info("Starting motion detection for camera %s.", self.id)
+
+    def detect_new_image_file(self):
+        """Method implement filesystem observer to detect new images in selected folder tree.
+           Only for ftp server camera notifications."""
+ 
+        logger.info("Starting filesystem observer for camera %s.", self.id)
```
[the remark this answers is not in the document]
Comment (stefanodellosa-personal) [reply src/domain/camera.py:35]: Ir's first commit, other will come to complete logic.

File: demo/motion_detection.py [hunk 10]
```diff
@@ -0,0 +1,90 @@
+import time
+import cv2
+from cv2_enumerate_cameras import enumerate_cameras
```
[the remark this answers is not in the document]
Comment (stefanodellosa-personal) [reply demo/motion_detection.py:3]: File will be removed

File: src/domain/select_local_cameras.py (hunk 3 above)
Comment (stefanodellosa-personal) [reply src/domain/select_local_cameras.py]: what's the difference? I would like to uniform the approach...

**Comment by stefanodellosa-personal** [comment]
Camera configuration UI added. Animal Detection mode and camera motion detection method added.

**Pull request merged**